department-of-veterans-affairs/va.gov-cms
# Specify default JSON:API Extras enhancers link fields globally

## Description

Closes #17021 

## Testing done

Manually tested locally and on Tugboat. Added a functional unit test.

## Screenshots

<img width="953" alt="Screenshot 2024-02-28 at 12 07 40 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3640707/d8b2d181-9c29-402e-8d09-b6212173934e">

## QA steps

1. Log into the site and head to "/admin/config/services/jsonapi/field_types".
   - [ ] Confirm that the "link" field type has the "URL for link (link field only)" enhancer applied to it.
   - [ ] "Get Absolute Urls" should be unchecked.
1. Navigate to "/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204"
   - [ ] Validate that `field_default_link` has a relative `url` with a value similar to the `uri` field's value.
1. Navigate to "/admin/config/services/jsonapi/field_types", check the "Get Absolute Urls" setting, and save the form.
1. Navigate to "/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204"
   - [ ] Validate that `field_default_link` has an absolute `url` with a value similar to the `uri` field's value.
1. Edit "/admin/config/services/jsonapi/add/resource_types/paragraph/react_widget" to add the "URL for link (link field only)" enhancer to the "field_default_link" field. Make sure to uncheck "Get Absolute Urls".
1. Navigate to "/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204"
   - [ ] Validate that `field_default_link` has a relative `url` with a value similar to the `uri` field's value.

### Definition of Done


### Select Team for PR review
